include/CL/sycl/vec.hpp

```diff
@@ -143,6 +143,72 @@ class vec : public detail::small_array<DataType,
     return std::tuple_cat(flatten<V>(i)...);
   }
 
+public:
+  /** An accessor to the first variable of a small array
+   */
+  DataType& x(){
+    static_assert(NumElements >= 1, "can't access to vec[0] if NumElements < 1");
+    return (*this)[0];
+  }
+
+
+  /** An accessor to the second variable of a small array
+  */
+  DataType& y(){
+    static_assert(NumElements >= 2, "can't access to vec[1] if NumElements < 2");
+    return (*this)[1];
+  }
+
+
+  /** An accessor to the third variable of a small array
+  */
+  DataType& z(){
+    static_assert(NumElements >= 3, "can't access to vec[2] if NumElements < 3");
+    return (*this)[2];
+  }
+
+  /** An accessor to the third variable of a small array
+  */
+  DataType& w(){
+    static_assert(NumElements >= 4, "can't access to vec[3] if NumElements < 4");
+    return (*this)[3];
+  }
+
+  /** Accessors to access hex indexed members of a small array.
+   * There are two macros, one for 0-9, one for A-F.
+   */
+#define TRISYCL_DECLARE_S(x)							\
+  DataType& s##x() {							\
+    static_assert(NumElements >= (x + 1), "can't access to vec["#x"] if NumElements < "#x"+1"); \
```

Comment:
While you are working on it, would it possible to untabify `TRISYCL_DECLARE_S` and `TRISYCL_DECLARE_Sx` to have only spaces instead of tabs with some variable interpretations leading to chaotic alignment of the `\`?